cleanlab_studio/internal/api/api.py

```diff
@@ -706,5 +713,70 @@ async def tlm_get_confidence_score(
     return cast(JSONDict, res_json)
 
 
+def enrich_data(
+    tlm: TLM,
+    prompt: str,
+    data: pd.DataFrame,
+    regex: Union[str, re.Pattern, List[re.Pattern]] = None,
+    return_values: List[str] = None,
+    subset_indices: Union[Tuple[int, int], List[int], None] = (0, 3),
+    column_name_prefix: str = "",
+    **kwargs,
+) -> pd.DataFrame:
+    """
+    Enriches a DataFrame with TLM responses and metadata.
+
+    Args:
+        tlm: Trustworthy Language Model instance.
+        prompt: Formatted f-string, that contains both the prompt, and names of columns to embed:
+        regex: One or more expressions will be passed into re.compile or a list of already compiled regular expressions.
+            If a list is proivded, the regexes are applied in order and first succesfull match is returned.
+        return_values: List of possible values to return (zero shot classification)
+            If specified, this only ever returns one of these values in the metadata column.
+            After your regex is applied, there may be additional transformations applied to ensure the returned value is one of these.
+        subset_indices: What subset of the supplied data to run this for. Can be either a list of unique indicies or a range. If None, we run on all the data.
+        column_name_prefix: Optional prefix appended to all columns names that are returned.
+
+    Returns:
+        A DataFrame that now contains additional `metadata` and `trustworthiness` columns related to the prompt. Columns will have `column_name_prefix_` prepended to them if specified.
+    """
+    subset_data = extract_df_subset(data, subset_indices)
+    formatted_prompts = subset_data.apply(lambda x: prompt.format(**x), axis=1).to_list()
+    outputs = tlm.try_prompt(formatted_prompts)
+
+    if column_name_prefix != "":
+        column_name_prefix = column_name_prefix + "_"
+
+    if (
+        regex is None and return_values is None
+    ):  # we do not need to have a "logs" column as original output is not augmented by regex or return values
+        subset_data[f"{column_name_prefix}metadata"] = [output["response"] for output in outputs]
+        subset_data[f"{column_name_prefix}trustworthiness"] = [
+            output["trustworthiness_score"] for output in outputs
+        ]
+        return subset_data
+
+    subset_data[f"{column_name_prefix}logs"] = [output["response"] for output in outputs]
+    subset_data[f"{column_name_prefix}trustworthiness"] = [
+        output["trustworthiness_score"] for output in outputs
+    ]
+
+    if regex:
+        regex_list = get_compiled_regex_list(regex)
+        subset_data[f"{column_name_prefix}metadata"] = subset_data[
+            f"{column_name_prefix}logs"
+        ].apply(lambda x: get_regex_match(x, regex_list))
```

Comment:
in cases where the regex didn't apply to a single example, can we print:

Your provided regex: {regex} did not match a single LLM output across the dataset. This message is simply to inform you that the regex is not having any effect.